behave/contrib/scenario_autoretry.py

```diff
@@ -48,17 +49,28 @@ def patch_scenario_with_autoretry(scenario, max_attempts=3):
 
     :param scenario:        Scenario or ScenarioOutline to patch.
     :param max_attempts:    How many times the scenario can be run.
+    :param delay:           Time to wait between scenario retries.
+    :param backoff:         Backoff factor to increase the delay per retry.
     """
     def scenario_run_with_retries(scenario_run, *args, **kwargs):
+        _delay = delay
+        delay_message = ""
         for attempt in range(1, max_attempts+1):
             if not scenario_run(*args, **kwargs):
                 if attempt > 1:
                     message = u"AUTO-RETRY SCENARIO PASSED (after {0} attempts)"
                     print(message.format(attempt))
                 return False    # -- NOT-FAILED = PASSED
+
             # -- SCENARIO FAILED:
             if attempt < max_attempts:
-                print(u"AUTO-RETRY SCENARIO (attempt {0})".format(attempt))
+                if delay:
+                    if attempt > 1: _delay *= backoff
+                    delay_message =  u" Delaying retry for {0}s".format(_delay)
+
+                print(u"AUTO-RETRY SCENARIO (attempt {0}){1}".format(attempt, delay_message))
```

Comment:
This is a bit hacky. A user will be required to start the `delay_message` with a blank character to make the output display decently.

Reply:
Can you elaborate? The user wouldn't be setting the `delay_message` here, it's added onto the current message which is printed. Nothing extra will be printed if the user hasn't added a delay.

Reply:
I understand. I wonder if there was a more elegant way to code this. :thinking: